lib/IMAP/MessageMapper.php

```diff
@@ -162,6 +167,7 @@ public function findAll(Horde_Imap_Client_Socket $client,
 				'ids' => new Horde_Imap_Client_Ids($lower . ':' . $upper)
```

Comment:
I'm still wondering whether we could directly fetch the messages themselves here, saving us another fetch / search query further down with `findByIds`

Reply:
In a fragmented mailbox you will run into overfetching when the target range of UIDs is dense. This will then create backpressure on how much data we need to download from IMAP *and* create a unexpectedly large number of INSERTs unless we discard the overfetched data.

So yes it is one IMAP roundtrip more right now but it prevents the overall synchronization from running too long and hitting a timeout.